src/backend/distributed/transaction/backend_data.c

```diff
@@ -476,6 +516,8 @@ AssignDistributedTransactionId(void)
 	int localGroupId = GetLocalGroupId();
 	TimestampTz currentTimestamp = GetCurrentTimestamp();
 
+	LockBackendSharedMemory(LW_EXCLUSIVE);
```

Comment:
Well, this prevents two concurrent distributed transactions to start. That's bad. Shall I swap `LW_EXCLUSIVE ` and `LW_SHARED`? That probably makes more sense